### PR TITLE
Refresh when content blocker is toggled in settings

### PR DESCRIPTION
**Pull request by subsymbolic**
Reviewer: Chris
Asana: https://app.asana.com/0/414709148257752/433776422524752

**Description**:
Update the site rating and tab to refresh when the content blocker toggle changes

**Steps to test this PR**:
TEST 1:
1. Navigate to a page
2. Go to settings, toggle the content blocker option off
3. Ensure that the page is refreshed and the site rating and content blocker views are updated to indicate that the content blocker is off

TEST 2:
1. Navigate to a page
2. Go to settings, toggle the content blocker option on
3. Ensure that the page is refreshed and the site rating and content blocker views are updated to indicate that the content blocker is on

###### Reviewer Checklist:
- [ ] **Ensure the PR solves the problem**
- [ ] **Review every line of code**
- [ ] **Ensure the PR does no harm by testing the changes thoroughly**
- [ ] **Get help if you're uncomfortable with any of the above!**
- [ ] Determine if there are any quick wins that improve the implementation

###### PR DRI Checklist:
- [x] Get advice or leverage existing code
- [x] Agree on technical approach with reviewer (if the changes are nuanced)
- [x] Ensure that there is a testing strategy (and documented non-automated tests)
- [x] Ensure there is a documented monitoring strategy (if necessary)
- [x] Consider systems implications (Database connections, Grafana stats, CPU)

### CODE REVIEW

File: Core/ContentBlockerConfigurationUserDefaults.swift [hunk 1]
```diff
@@ -71,13 +72,17 @@ public class ContentBlockerConfigurationUserDefaults: ContentBlockerConfiguratio
...
     
     public func removeFromWhitelist(domain: String) {
         var whitelist = domainWhitelist
         whitelist.remove(domain)
         domainWhitelist = whitelist
-        SFContentBlockerManager.reloadContentBlocker()
+        onStoreChanged()
+    }
+    
+    private func onStoreChanged() {
```
Comment (brindy) [inline Core/ContentBlockerConfigurationUserDefaults.swift]: `notifiyStoreChanged` might be a better name for this?  `onXXX` is usually the method name in the receiver of such events.

File: DuckDuckGo/SiteRatingView.swift [hunk 2]
```diff
@@ -38,27 +38,19 @@ public class SiteRatingView: UIView {
...
-    
-    private func addUserDefaultsObserver() {
-        NotificationCenter.default.addObserver(self, selector: #selector(onUserDefaultsChanged), name: UserDefaults.didChangeNotification, object: nil)
-    }
-    
-    private func removeUserDefaultsObserver() {
-        NotificationCenter.default.removeObserver(self, name: UserDefaults.didChangeNotification, object: nil)
+
+    private func addContentBlockerConfigurationObserver() {
+        NotificationCenter.default.addObserver(self, selector: #selector(onContentBlockerConfigurationChanged), name: ContentBlockerConfigurationChangedNotification.name, object: nil)
```
Comment (brindy) [inline DuckDuckGo/SiteRatingView.swift]: This observer never seems to get removed.

File: DuckDuckGo/TabViewController.swift [hunk 3]
```diff
@@ -43,12 +43,25 @@ class TabViewController: WebViewController {
...
+        addContentBlockerConfigurationObserver()
+    }
     
     override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
         resetNavigationBar()
     }
     
+    private func addContentBlockerConfigurationObserver() {
+        NotificationCenter.default.addObserver(self, selector: #selector(onContentBlockerConfigurationChanged), name: ContentBlockerConfigurationChangedNotification.name, object: nil)
```
Comment (brindy) [inline DuckDuckGo/TabViewController.swift]: This observer never seems to get removed either.

File: DuckDuckGo/SiteRatingView.swift [hunk 4]
```diff
@@ -38,27 +38,19 @@ public class SiteRatingView: UIView {
         let view = Bundle.main.loadNibNamed("SiteRatingView", owner: self, options: nil)![0] as! UIView
         self.addSubview(view)
         view.frame = self.bounds
-        addUserDefaultsObserver()
-    }
-    
-    deinit {
-        removeUserDefaultsObserver()
+        addContentBlockerConfigurationObserver()
```
Comment (brindy) [inline DuckDuckGo/SiteRatingView.swift]: Does the `SiteRatingView` need to observe this? If the page is reloaded won't it update anyway via `webpageDidFinishLoading` in the view controller?

File: Core/ContentBlockerConfigurationUserDefaults.swift (hunk 1 above)
Comment (brindy) [reply Core/ContentBlockerConfigurationUserDefaults.swift]: Actually, can see a couple of places where this convention is used - feel free to ignore.

File: DuckDuckGo/SiteRatingView.swift (hunk 4 above)
Comment (subsymbolic) [reply DuckDuckGo/SiteRatingView.swift]: The reload can take a long time so we want to update it immediately. We don't the user staring at an exclamation mark which suggests that the content blocker is off when they have switched it back on.

File: DuckDuckGo/SiteRatingView.swift (hunk 2 above)
Comment (subsymbolic) [reply DuckDuckGo/SiteRatingView.swift]: That's intentional. As of iOS 8 you only remove them manually if you want to stop observing early. See the discussion in https://developer.apple.com/documentation/foundation/notificationcenter/1415360-addobserver.

File: DuckDuckGo/TabViewController.swift (hunk 3 above)
Comment (subsymbolic) [reply DuckDuckGo/TabViewController.swift]: Ad above.

**Pull request merged**